### PR TITLE
Fix regex password mask

### LINKED ISSUE #195: CAS logs passwords in log level debug

### PR DESCRIPTION
**Pull request by nhinze23**
An invalid regex was used to mask the password.
The output changed from simple string 'password' to a char array 'array<Character>[p, a , s , s , w , o , r , d]'

Resolves #195

### CODE REVIEW

File: app/src/main/java/de/triology/cas/logging/DefaultMappingPasswordRewritePolicy.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ protected String getPasswordFlag() {
     protected String replacePasswordValue(String originMessage) {
         String modifiedMessage = null;
         if (originMessage != null) {
-            modifiedMessage = originMessage.replaceAll("mappedValue = '\\S+'", "mappedValue = '****'");
+            modifiedMessage = originMessage.replaceAll("mappedValue = array<Character>\\[.*\\]", "mappedValue = '****'");
```
Comment (meiserloh) [inline app/src/main/java/de/triology/cas/logging/DefaultMappingPasswordRewritePolicy.java:36]: Why can we not just use a generic solution like this?  
```suggestion
            modifiedMessage = originMessage.replaceAll("mappedValue = .*", "mappedValue = '****'");
```

Comment (nhinze23) [reply app/src/main/java/de/triology/cas/logging/DefaultMappingPasswordRewritePolicy.java:36]: With this change you will delete all chars after the mappedValue.

Comment (meiserloh) [reply app/src/main/java/de/triology/cas/logging/DefaultMappingPasswordRewritePolicy.java:36]: yeah, I just noticed that, too... But there has to be a solution to this right? I mean it's regex :D

Comment (meiserloh) [inline app/src/main/java/de/triology/cas/logging/DefaultMappingPasswordRewritePolicy.java:36]: A second thing: This regex would not match the line in the supportticket, would it?
"mappedValue = array[T, e, s, t]" does not contain the `<Character>` part.

Comment (ppxl) [reply app/src/main/java/de/triology/cas/logging/DefaultMappingPasswordRewritePolicy.java:36]: I didn't get this one but I do concur in that it seems that we do not need this line at all.

Thus I suggest something along the lines of this so we do not have to build pattern matching during runtime

```java
public final class DefaultMappingPasswordRewritePolicy extends AbstractCASPasswordRewritePolicy {
    private static final String PARAMETER_PASSWORD_TEXT = "parameter:'password'";
    private static final Pattern passwordDebugLogLine = Pattern.compile("^.+" + PARAMETER_PASSWORD_TEXT + ".+$");
    private static final String replaceMessage = "this log line contained sensitive data and was removed";

// later in this class

    protected String replacePasswordValue(String originMessage) {
        if (originMessage == null || originMessage.isEmpty()) {
            return null;
        }

        boolean logLineContainsPassword = passwordDebugLogLine.matcher(originMessage).find();
        if (logLineContainsPassword) {
            return replaceMessage;
        }

        return originMessage;
    }
```

**Review by ppxl — COMMENTED** [review]
Thank you for your quick work :muscle: If we really do not need the log line at all I suggest to remove it in fullness. The only fixed part over the course of CAS 6 and 7 is the string `parameter:'password` which I would designate as event for log line deletion.

File: app/src/main/java/de/triology/cas/logging/DefaultMappingPasswordRewritePolicy.java (hunk 1 above)
Comment (nhinze23) [reply app/src/main/java/de/triology/cas/logging/DefaultMappingPasswordRewritePolicy.java:36]: I think that is good too. But we do not have to match the `PARAMETER_PASSWORD_TEXT` again. The `AbstractCASPasswordRewritePolicy` checks this with `containsPassword(LogEvent source)`

**Review by ppxl — APPROVED** [review]
LGTM

**Pull request merged**